### ISSUE TITLE
The ordered deployment of Helm charts for Istio pattern is broken with latest commit

### ISSUE THREAD

**Issue by vchintal** [aws-ia/terraform-aws-eks-blueprints#1766]
## Description

In the file here https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/main/patterns/istio/main.tf because of the [latest commit](https://github.com/aws-ia/terraform-aws-eks-blueprints/commit/2d96b786accf89547351cc8fef8560f553336b23#diff-83420a756427c465ea1934de7bda6f08979274a243460cffc8425b864e90e9f5)

1. The Helm chart installs that were done outside of the add-ons have been moved under EKS add-ons
2. The Helm chart dependencies that were previously met are now lost as they (charts) all seem to be executing in parallel and because of this, `istio-ingress` is not deploying correctly after `istiod`, as per the expectation.

### Expected Behavior

`istio-ingress` helm chart is run after the `istiod` helm chart, in that order. 

### Actual Behavior

`istio-ingress` helm chart sometimes seems to run before the `istiod` helm chart

## Suggested Correction

As the Helm charts for Istio do not belong to the EKS add-ons, it is recommended that they be restored back and run as Terraform `helm_release` resources.

**Comment by bryantbiggs** [comment]
could you further clarify or demonstrate whats broken or the errors you are seeing?

If I was a user who deploys Istio using a GitOps operator instead of Terraform, how would I approach this situation you are describing?

**Comment by vchintal** [comment]
Issue 1:

In a clean run I face the following issue:
 
<img width="2544" alt="image" src="https://github.com/aws-ia/terraform-aws-eks-blueprints/assets/4762573/b2affc3d-e1be-416d-b88f-b25f74afe065">

1. Looking at the code, I can only imagine this happening because of this `istiod` Helm release [line](https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/main/patterns/istio/main.tf#L136). 
2. But for `istio-base` Helm release the [line](https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/main/patterns/istio/main.tf#L127) has the `create_namespace = true` for `istio-system`. 
4. So clearly the  `helm_releases` are not running sequentially as perhaps hoped

**Comment by bryantbiggs** [comment]
Should we create the shared namespace through a standalone `kubernetes_namespace_v1` resource and pass that in to the two charts that are deployed into that namespace?

**Comment by vchintal** [comment]
Issue 2:

After setting `create_namespace = true` in [line](https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/main/patterns/istio/main.tf#L136) and rerunning the Terraform and all the subsequent instructions in the README we see the following with `istio-ingress` pod in `ImagePullBackOff` status. This is new issue we had not seen before.

<img width="2532" alt="image" src="https://github.com/aws-ia/terraform-aws-eks-blueprints/assets/4762573/7bde14f9-e063-431c-90c9-020702034eb9">

```
5m4s        Normal    Pulling                        pod/istio-ingress-6f7c5dffd8-59nz8      Pulling image "auto"
5m4s        Warning   Failed                         pod/istio-ingress-6f7c5dffd8-59nz8      Failed to pull image "auto": rpc error: code = Unknown desc = failed to pull and unpack image "docker.io/library/auto:latest": failed to resolve reference "docker.io/library/auto:latest": pull access denied, repository does not exist or may require authorization: server message: insufficient_scope: authorization failed
5m4s        Warning   Failed                         pod/istio-ingress-6f7c5dffd8-59nz8      Error: ErrImagePull
83s         Normal    BackOff                        pod/istio-ingress-6f7c5dffd8-59nz8      Back-off pulling image "auto"
4m48s       Warning   Failed                         pod/istio-ingress-6f7c5dffd8-59nz8      Error: ImagePullBackOff
```

**Comment by vchintal** [comment]
> I don't think we can have two release definitions that are both trying to create the same namespace, what are your thoughts about the comment above [#1766 (comment)](https://github.com/aws-ia/terraform-aws-eks-blueprints/issues/1766#issuecomment-1725890313)

Perhaps true but the issue is not with namespaces but having `istiod` run before `istio-ingress`.
